homeassistant/components/lg_soundbar/config_flow.py

```diff
@@ -0,0 +1,76 @@
+"""Config flow to configure the LG Soundbar integration."""
+import temescal
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT, CONF_UNIQUE_ID
+
+from .const import DEFAULT_HOST, DEFAULT_PORT, DOMAIN
+
+DATA_SCHEMA = {
+    vol.Required(CONF_HOST, default=DEFAULT_HOST): str,
+    vol.Required(CONF_PORT, default=DEFAULT_PORT): vol.Coerce(int),
+    vol.Optional(CONF_NAME): str,
+}
+
+
+class LGSoundbarConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """LG Soundbar config flow."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        errors = {}
+        if user_input is None:
+            return await self._show_form(user_input)
+        if not errors:
+            try:
+                uuid = self.test_connect(user_input[CONF_HOST], user_input[CONF_PORT])
+            except ConnectionError:
+                errors["base"] = "cannot_connect"
+            else:
+                await self.async_set_unique_id(uuid)
+                self._abort_if_unique_id_configured()
+                info = {
+                    CONF_UNIQUE_ID: uuid,
```

Comment:
Why store this ? It's also already stored as unique ID of the config entry.

Reply:
that is passed to the media_player, how else would i create the opject that the media player is exspecting? is there a way to pull the unique ID of the config entry from inside the media_player?

Reply:
we pass the config entry to `async_setup_entry` in `media_player.py`